src/nprintml/net/step.py

```diff
@@ -153,7 +153,26 @@ def make_output_directory(cls, args):
         outdir.mkdir()
         return outdir
 
-    def generate_argv(self, args, pcap_file=None):
+    @staticmethod
+    def make_output_path(outdir, pcap_file, dir_basis):
+        if not pcap_file:
+            return None
+
+        npt_file = pathlib.Path(pcap_file).with_suffix('.npt')
+        if dir_basis:
+            npt_file = npt_file.relative_to(dir_basis)
+        else:
+            npt_file = npt_file.name
+
+        npt_path = outdir / npt_file
+
+        if npt_path.exists():
+            raise FileExistsError(None, 'nPrint output path collision', str(npt_path))
+
+        npt_path.parent.mkdir(parents=True, exist_ok=True)
+        return npt_path
```

Comment:
This and the rest is just about recursing an input directory tree and preserving that tree in the nPrint output directory.